src/fx.js

```diff
@@ -214,45 +214,28 @@ SVG.FX = SVG.invent({
     // updates all animations to the current state of the element
     // this is important when one property could be changed from another property
   , initAnimations: function() {
-      var i
+      var i, source
       var s = this.situation
 
       if(s.init) return this
 
       for(i in s.animations){
+        source = this.target()[i]()
 
-        if(i == 'viewbox'){
-          s.animations[i] = this.target().viewbox().morph(s.animations[i])
-        }else{
-
-          // TODO: this is not a clean clone of the array. We may have some unchecked references
-          s.animations[i].value = (i == 'plot' ? this.target().array().value : this.target()[i]())
-
-          // sometimes we get back an object and not the real value, fix this
-          if(s.animations[i].value.value){
-            s.animations[i].value = s.animations[i].value.value
-          }
-
-          if(s.animations[i].relative)
-            s.animations[i].destination.value = s.animations[i].destination.value + s.animations[i].value
-
-        }
+        // The condition is because some methods return a normal number instead
+        // of a SVG.Number
+        if(s.animations[i] instanceof SVG.Number)
+          source = new SVG.Number(source)
```

Comment:
Actually, the test here check if the _destination_ is an SVG.Number (`s.animations[i]` is first set in the method `x`, `y`, etc... of the FX module. If it is, that mean that the _source_ must also be an SVG.Number so it is converted to one.